### PR TITLE
Feature/csv export

### PR DESCRIPTION
**Pull request by kolyaratishin**
I made it possible to export to csv (#98)

### CODE REVIEW

**Review by VolodymyrHoliev — APPROVED** [review]

File: api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java [hunk 1]
```diff
@@ -2,68 +2,106 @@
...
-        return new CsvParser(settings);
-    }
-
     @Test
     @SneakyThrows
     void dataImport_isOk() {
+        when(rowProcessor.getBeans()).thenReturn(Arrays.asList(
+                new MedicineDto("Aspirin", BigDecimal.valueOf(25.25), "extID", "ANC"),
+                new MedicineDto("Fervex", BigDecimal.valueOf(25.25), "extID", "Liki24")
+        ));
```
Comment (Anna-Kudiakova) [inline api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java:104]: these changes are not relevant to the topic of the pull request

**Review by Anna-Kudiakova — APPROVED** [review]

File: api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java (hunk 1 above)
Comment (kolyaratishin) [reply api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java:104]: This test and this testClass was wrote bad and i had to change some fields and methods in order to fix it. Also, this test failed when I did the PR, so it had to be rewritten

File: api/src/test/resources/export/export.csv [hunk 2]
```diff
@@ -0,0 +1,7 @@
+title,price,pharmacyName
+Aspirin,25.25,ANC
+Aspirin,35.25,3I
+Aspirin,45.25,Liki24
+Fervex,22.25,ANC
+Fervex,32.25,3I
+Fervex,42.25,Liki24
```
Comment (MuchosChuchos) [inline api/src/test/resources/export/export.csv:7]: add blank line. =)

**Review by MuchosChuchos — CHANGES_REQUESTED** [review]

**Comment by VolodymyrHoliev** [comment]
@Anna-Kudiakova since you've approved this PR could you please resolve the conversations above?

File: api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java (hunk 1 above)
Comment (Oleksandr-Eleks) [reply api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java:104]: >_<
I'm a bit confused that there is no real import.csv file.
Please test it on a real small file.

File: api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java [hunk 3]
```diff
@@ -82,4 +120,50 @@ void dataImport_isInvalidFormat() {
...
+    @SneakyThrows
+    void dataExport_isOk() {
+        String filepath = "src/test/resources/export/export.csv";
+        File file = new File(filepath);
+
+        when(pharmacyRepository.findAll()).thenReturn(pharmacies);
+        pharmacies.forEach(pharmacy ->
+                when(pharmacyRepository.findById(pharmacy.getId())).thenReturn(Optional.of(pharmacy)));
+        when(priceRepository.findAllMedicinesPrices()).thenReturn(prices);
+
```
Comment (Oleksandr-Eleks) [inline api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java:133]: Please remove BL

File: api/src/test/java/com/eleks/academy/pharmagator/controllers/ExportControllerIT.java [hunk 4]
```diff
@@ -84,10 +89,34 @@ void exportToPdf_getPdfWriterIsBad() {
     void exportToPdf_createFontIsInvalid() {
         when(pdfExportService.export()).thenThrow(new ExportExceptions(ExportExceptions.Error.INVALID_FONT));
 
-        this.mockMvc.perform(MockMvcRequestBuilders.get(URI))
+        this.mockMvc.perform(MockMvcRequestBuilders.get(PDF_URI))
```
[the remark this answers is not in the document]
Comment (Oleksandr-Eleks) [reply api/src/test/java/com/eleks/academy/pharmagator/controllers/ExportControllerIT.java:92]: The same as prev.

**Review by MuchosChuchos — APPROVED** [review]

File: api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java (hunk 3 above)
Comment (kolyaratishin) [reply api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java]: done

File: api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java (hunk 1 above)
Comment (kolyaratishin) [reply api/src/test/java/com/eleks/academy/pharmagator/services/CsvServiceImplTest.java:104]: It will be fixed in next PR

**Pull request merged**